jpa/hibernate6/src/main/java/org/jboss/as/jpa/hibernate/management/QueryName.java

```diff
@@ -0,0 +1,216 @@
+    private static final String DIV = "/";
+    private static final String MOD = "%";
+    private static final String COLON = ":";
+    private static final String PARAM = "?";
+    private static final String COMMA = ",";
+    private static final String SPACE = " ";
+    private static final String TAB = "\t";
+    private static final String NEWLINE = "\n";
+    private static final String LINEFEED = "\r";
+    private static final String QUOTE = "'";
+    private static final String DQUOTE = "\"";
+    private static final String TICK = "`";
+    private static final String OPEN_BRACE = "{";
+    private static final String CLOSE_BRACE = "}";
+    private static final String HAT = "^";
+    private static final String AMPERSAND = "&";
+
+    // textual representation (not to be localized as we don't won't duplication between any of the values)
+    private static final String NOT_EQUAL__ = "_not_equal_";
+    private static final String BANG_NOT_EQUAL__ = "_bang_not_equal_";
+    private static final String HAT_NOT_EQUAL__ = "_hat_not_equal_";
+    private static final String LESS_THAN_EQUAL__ = "_less_than_equal_";
+    private static final String GREATER_THAN_EQUAL__ = "_greater_than_equal_";
+    private static final String CONCAT__ = "_concat_";
+    private static final String LESS_THAN__ = "_less_than_";
+    private static final String EQUAL__ = "_equal_";
+    private static final String GREATER__ = "_greater_";
+    private static final String LEFT_PAREN__ = "_left_paren_";
+    private static final String RIGHT_PAREN__ = "_right_paren_";
+    private static final String LEFT_BRACKET__ = "_left_bracket_";
+    private static final String RIGHT_BRACKET__ = "_right_bracket_";
+    private static final String PLUS__ = "_plus_";
+    private static final String MINUS__ = "_minus_";
+    private static final String STAR__ = "_star_";
+    private static final String DIVIDE__ = "_divide_";
+    private static final String MODULUS__ = "_modulus_";
+    private static final String COLON__ = "_colon_";
+    private static final String PARAM__ = "_param_";
+    private static final String COMMA__ = "_comma_";
+    private static final String SPACE__ = "_space_";
+    private static final String TAB__ = "_tab_";
+    private static final String NEWLINE__ = "_newline_";
+    private static final String LINEFEED__ = "_linefeed_";
+    private static final String QUOTE__ = "_quote_";
+    private static final String DQUOTE__ = "_double_quote_";
+    private static final String TICK__ = "_tick_";
+    private static final String OPEN_BRACE__ = "_left_brace_";
+    private static final String CLOSE_BRACE__ = "_right_brace_";
+    private static final String HAT__ = "_hat_";
+    private static final String AMPERSAND__ = "_ampersand_";
+
+    public static QueryName queryName(String query) {
+        return new QueryName(query);
+    }
+
+    /**
+     * Construct
+     *
+     * @param query
+     */
+    public QueryName(String query) {
```

Comment:
Maybe it's intended: A static factory method but the constructor is public. Is a private constructor needed here?

Reply:
Thanks @boris-unckel , just switched to a private constructor